lib/trie/lookup.go

```diff
@@ -0,0 +1,87 @@
+package trie
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/ChainSafe/chaindb"
+)
+
+var (
+	// ErrProofNodeNotFound when a needed proof node is not in the database
+	ErrProofNodeNotFound = errors.New("cannot found the state root in storage")
+)
+
+// Lookup struct holds the state root and database reference
+// used to retrieve trie information from database
+type Lookup struct {
```

Comment:
the types/funcs in this file can be unexported, as they aren't used by any other package